<a href="https://colab.research.google.com/github/BobbyLeeSH/ML_STUDY/blob/main/tf_lemonade.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 생활코딩 tensorflow 101 수업을 들으면서 따라해보는 코드입니다.
## Lemonade sales prediction

# Import pandas and tensorflow

In [ ]:
import pandas as pd
import tensorflow as tf

In [ ]:
# read data
data_path = 'https://raw.githubusercontent.com/blackdew/tensorflow1/master/csv/lemonade.csv'
lemonade = pd.read_csv(data_path)

print(lemonade.head())
print(lemonade.shape)
print(lemonade.columns)

   온도  판매량
0  20   40
1  21   42
2  22   44
3  23   46
4  24   48
(6, 2)
Index(['온도', '판매량'], dtype='object')


In [ ]:
# separate features and label
feature = lemonade[['온도']]
label = lemonade[['판매량']]

In [ ]:
# build model using Neural Network
X = tf.keras.layers.Input(shape=[1])
Y = tf.keras.layers.Dense(1)(X)
model = tf.keras.models.Model(X,Y)
model.compile(loss='mse')


```
X = tf.keras.layers.Input(shape=[1])
```
means we are using one Feature


```
Y = tf.keras.layers.Dense(1)(X)
```
means we are using one Label

In [ ]:
model.fit(feature, label, epochs=1000, verbose=0)
model.fit(feature, label, epochs=10)

Epoch 1/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0040
Epoch 2/10
1/1 [==============================] - 0s 3ms/step - loss: 0.0041
Epoch 3/10
1/1 [==============================] - 0s 3ms/step - loss: 0.0041
Epoch 4/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0041
Epoch 5/10
1/1 [==============================] - 0s 5ms/step - loss: 0.0041
Epoch 6/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0041
Epoch 7/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0041
Epoch 8/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0040
Epoch 9/10
1/1 [==============================] - 0s 3ms/step - loss: 0.0040
Epoch 10/10
1/1 [==============================] - 0s 4ms/step - loss: 0.0040


In [ ]:
# use model to predict
print(model.predict([[15]]))

[[30.284992]]


### it is predicted that 30.284992 lemonade will be sold if the temperature is 15 degree celcius